### PR TITLE
automatic pr update workflow removing mergepal

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- removes mergepal
- adds workflow to automatically update prs which are marked as "auto merge" (using the github PR feature)
- we will not use the `ready-to-merge` label anymore

**Status (please check what you already did)**:
- [x] updated the documentation

Once merged: announce in engineering

### CODE REVIEW

**Comment by tmbo** [comment]
I've tried this on a test repo and it looked good. Nevertheless, there might be cases where we don't update a PR. Since risk is low, we should merge + keep an open eye and fix as we go.

File: .github/workflows/automatic-pr-update.yml [hunk 1]
```diff
@@ -2,24 +2,16 @@ name: Automatic PR Merger
...
 
 jobs:
   # thats's all. single step is needed - if PR is mergeable according to
   # branch protection rules it will be merged automatically
   mergepal:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: rasahq/merge-pal-action@master
+      - uses: actions/checkout@v1
```
Comment (wochinge) [inline .github/workflows/automatic-pr-update.yml:12]:
```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/automatic-pr-update.yml [hunk 2]
```diff
@@ -2,24 +2,16 @@ name: Automatic PR Merger
...
+      - uses: rasahq/update-pr-branch@601ac3dadf708b14e7f0252d98230538235f99d9
+        if: github.repository == 'RasaHQ/rasa'
         with:
           token: ${{ secrets.RASABOT_AUTOMERGE_GITHUB_TOKEN || secrets.GITHUB_TOKEN}}
+          required_approval_count: 0
```
Comment (wochinge) [inline .github/workflows/automatic-pr-update.yml:17]: is this a workaround or why do we need it?

Comment (wochinge) [reply .github/workflows/automatic-pr-update.yml:17]: if it's a workaround then I'd vote to add a comment

**Review by wochinge — APPROVED** [review]

File: .github/workflows/automatic-pr-update.yml (hunk 2 above)
Comment (tmbo) [reply .github/workflows/automatic-pr-update.yml:17]: yes it is - this is an odd requirement from the original plugin. I could have changed it in the forked actions code but I wanted the changes to stay minimal.

Comment (wochinge) [reply .github/workflows/automatic-pr-update.yml:17]: makes complete sense 👍🏻

File: .github/workflows/automatic-pr-update.yml [hunk 3]
```diff
@@ -20,6 +10,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: rasahq/merge-pal-action@master
+      - uses: rasahq/update-pr-branch@601ac3dadf708b14e7f0252d98230538235f99d9
+        if: github.repository == 'RasaHQ/rasa'
         with:
           token: ${{ secrets.RASABOT_AUTOMERGE_GITHUB_TOKEN || secrets.GITHUB_TOKEN}}
+          required_approval_count: 0
```
Comment (tmbo) [inline .github/workflows/automatic-pr-update.yml:17]:
```suggestion
          # required parameter by original action - 
          # check is already done through protected branches so not needed for us
          required_approval_count: 0   
```

File: .github/workflows/automatic-pr-update.yml [hunk 4]
```diff
@@ -20,6 +10,10 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
-      - uses: rasahq/merge-pal-action@master
+      - uses: rasahq/update-pr-branch@601ac3dadf708b14e7f0252d98230538235f99d9
+        if: github.repository == 'RasaHQ/rasa'
         with:
           token: ${{ secrets.RASABOT_AUTOMERGE_GITHUB_TOKEN || secrets.GITHUB_TOKEN}}
+          # required parameter by original action - 
+          # check is already done through protected branches so not needed for us
+          required_approval_count: 0
```
Comment (virtualroot) [inline .github/workflows/automatic-pr-update.yml:19]: Just me being picky 😅

```suggestion
          required_approval_count: 0
```

**Review by virtualroot — APPROVED** [review]

File: .github/workflows/automatic-pr-update.yml [hunk 5]
```diff
@@ -20,6 +10,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - uses: rasahq/merge-pal-action@master
+      - uses: rasahq/update-pr-branch@601ac3dadf708b14e7f0252d98230538235f99d9
+        if: github.repository == 'RasaHQ/rasa'
         with:
           token: ${{ secrets.RASABOT_AUTOMERGE_GITHUB_TOKEN || secrets.GITHUB_TOKEN}}
```
Comment (virtualroot) [inline .github/workflows/automatic-pr-update.yml:16]: One more thing, documentation says action only works with [PAT](https://github.com/rasahq/update-pr-branch#token), we can remove `secrets.GITHUB_TOKEN`

```suggestion
          token: ${{ secrets.RASABOT_AUTOMERGE_GITHUB_TOKEN }}
```

File: .github/workflows/automatic-pr-update.yml (hunk 4 above)
Comment (tmbo) [reply .github/workflows/automatic-pr-update.yml:19]: very much appreciated :D - an obvious flaw of the in-browser editor

**Pull request merged**